### PR TITLE
syncer(dm): Ignore SET statement for MariaDB

### LINKED ISSUE #10206: DPANIC with MariaDB 10.6
### What did you do?

Setup DM with MariaDB 10.6.11

```
    [2023/11/29 11:17:55.229 +02:00]
    [DPANIC]
    [tracker.go:206]
    ["unexpected statement type"]
    [task=demo-mariadb-tidb]
    [unit="binlog replication"]
    [component=schema-tracker]
    [task=demo-mariadb-tidb]
    [type=*ast.SetStmt]
    [stack="github.com/pingcap/tiflow/dm/pkg/schema.(*Tracker).Exec\n\tgithub.com/pingcap/tiflow/dm/pkg/schema/tracker.go:206\ngithub.com/pingcap/tiflow/dm/syncer.(*Syncer).loadTableStructureFromDump\n\tgithub.com/pingcap/tiflow/dm/syncer/syncer.go:3068\ngithub.com/pingcap/tiflow/dm/syncer.(*Syncer).Run\n\tgithub.com/pingcap/tiflow/dm/syncer/syncer.go:1856\ngithub.com/pingcap/tiflow/dm/syncer.(*Syncer).Process\n\tgithub.com/pingcap/tiflow/dm/syncer/syncer.go:757"]
    
    [2023/11/29 11:17:55.231 +02:00]
    [INFO]
    [syncer.go:1688]
    ["received ungraceful exit ctx, exit now"]
    [task=demo-mariadb-tidb]
    [unit="binlog replication"]
    panic: interface conversion: ast.StmtNode is *ast.SetStmt, not *ast.CreateTableStmt
    
    goroutine 1807 [running]:
    github.com/pingcap/tiflow/dm/syncer.(*Syncer).loadTableStructureFromDump(0xc000cb7200, {0x4a54438, 0xc00105dd10})
            github.com/pingcap/tiflow/dm/syncer/syncer.go:3079 +0x1ce9
    github.com/pingcap/tiflow/dm/syncer.(*Syncer).Run(0xc000cb7200, {0x4a54438?, 0xc00105dd10?})
            github.com/pingcap/tiflow/dm/syncer/syncer.go:1856 +0xcd6
    github.com/pingcap/tiflow/dm/syncer.(*Syncer).Process(0xc000cb7200, {0x4a54438, 0xc00105dcc0}, 0x0?)
            github.com/pingcap/tiflow/dm/syncer/syncer.go:757 +0x325
    created by github.com/pingcap/tiflow/dm/worker.(*SubTask).run in goroutine 1707
            github.com/pingcap/tiflow/dm/worker/subtask.go:242 +0x53b
```

### What did you expect to see?

_No response_

### What did you see instead?

_No response_

### Versions of the cluster

Master @ 13499f4b366202cb46beba631a66997a62c99884

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

### PR DESCRIPTION
**Pull request by dveeden**
Without this the following happens:
```
[2023/11/29 11:17:55.229 +02:00]
[DPANIC]
[tracker.go:206]
["unexpected statement type"]
[task=demo-mariadb-tidb]
[unit="binlog replication"]
[component=schema-tracker]
[task=demo-mariadb-tidb]
[type=*ast.SetStmt]
[stack="github.com/pingcap/tiflow/dm/pkg/schema.(*Tracker).Exec\n\tgithub.com/pingcap/tiflow/dm/pkg/schema/tracker.go:206\ngithub.com/pingcap/tiflow/dm/syncer.(*Syncer).loadTableStructureFromDump\n\tgithub.com/pingcap/tiflow/dm/syncer/syncer.go:3068\ngithub.com/pingcap/tiflow/dm/syncer.(*Syncer).Run\n\tgithub.com/pingcap/tiflow/dm/syncer/syncer.go:1856\ngithub.com/pingcap/tiflow/dm/syncer.(*Syncer).Process\n\tgithub.com/pingcap/tiflow/dm/syncer/syncer.go:757"]

[2023/11/29 11:17:55.231 +02:00]
[INFO]
[syncer.go:1688]
["received ungraceful exit ctx, exit now"]
[task=demo-mariadb-tidb]
[unit="binlog replication"]
panic: interface conversion: ast.StmtNode is *ast.SetStmt, not *ast.CreateTableStmt

goroutine 1807 [running]:
github.com/pingcap/tiflow/dm/syncer.(*Syncer).loadTableStructureFromDump(0xc000cb7200, {0x4a54438, 0xc00105dd10})
	github.com/pingcap/tiflow/dm/syncer/syncer.go:3079 +0x1ce9
github.com/pingcap/tiflow/dm/syncer.(*Syncer).Run(0xc000cb7200, {0x4a54438?, 0xc00105dd10?})
	github.com/pingcap/tiflow/dm/syncer/syncer.go:1856 +0xcd6
github.com/pingcap/tiflow/dm/syncer.(*Syncer).Process(0xc000cb7200, {0x4a54438, 0xc00105dcc0}, 0x0?)
	github.com/pingcap/tiflow/dm/syncer/syncer.go:757 +0x325
created by github.com/pingcap/tiflow/dm/worker.(*SubTask).run in goroutine 1707
	github.com/pingcap/tiflow/dm/worker/subtask.go:242 +0x53b
```

### What problem does this PR solve?

Issue Number: close #10206 

#### Tests 

 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
Compatibility with MariaDB has been improved.
```

### CODE REVIEW

**Comment by dveeden** [comment]
Would be good to add some tests for this.

File: dm/syncer/syncer.go [hunk 1]
```diff
@@ -3075,10 +3075,15 @@ func (s *Syncer) loadTableStructureFromDump(ctx context.Context) error {
...
 				continue
 			}
-			s.saveTablePoint(
-				&filter.Table{Schema: db, Name: stmtNode.(*ast.CreateTableStmt).Table.Name.O},
-				s.getFlushedGlobalPoint(),
-			)
+			switch v := stmtNode.(type) {
+			case *ast.CreateTableStmt:
+				s.saveTablePoint(
+					&filter.Table{Schema: db, Name: stmtNode.(*ast.CreateTableStmt).Table.Name.O},
```
Comment (lance6716) [inline dm/syncer/syncer.go:3081]:
```suggestion
					&filter.Table{Schema: db, Name: v.Table.Name.O},
```

I remember we can `v` like this.

**Review by lance6716 — COMMENTED** [review]
rest lgtm

**Comment by lance6716** [comment]
ping @dveeden

File: dm/syncer/syncer.go [hunk 2]
```diff
@@ -3075,10 +3075,15 @@ func (s *Syncer) loadTableStructureFromDump(ctx context.Context) error {
...
+					s.getFlushedGlobalPoint(),
+				)
+			default:
+				logger.Warn("ignoring statement", zap.String("type", fmt.Sprintf("%T", v)))
```
[the remark this answers is not in the document]
Comment (dveeden) [reply dm/syncer/syncer.go]: Done in f6e7f4af221c7ea359b7f33384c6eef09a7eeb92

File: dm/pkg/schema/tracker.go [hunk 3]
```diff
@@ -202,8 +202,14 @@ func (tr *Tracker) Exec(ctx context.Context, db string, stmt ast.StmtNode) (errR
 	case *ast.TruncateTableStmt:
 		ident := ast.Ident{Schema: v.Table.Schema, Name: v.Table.Name}
 		return tr.upstreamTracker.TruncateTable(tr.se, ident)
+	case *ast.SetStmt:
```
[the remark this answers is not in the document]
Comment (dveeden) [reply dm/pkg/schema/tracker.go]: I've updated the code. PTAL

**Review by lance6716 — APPROVED** [review]

**Pull request merged**